### ISSUE TITLE
C API: Use of `nullptr` as default parameter causing failure to compile

### ISSUE THREAD

**Issue by andyf-canva** [thorvg/thorvg#2228]
We are trying to access the C API for use within a Swift Package, and the use of `nullptr` as a default parameter in the `tvg_animation_get_segment` function is preventing us from compiling the library. This is present in the `v0.13.2` release.

Here is [the specific line](https://github.com/thorvg/thorvg/blob/bb7f23da7865a1ff970ff3203ee9e883adc1ce4e/src/bindings/capi/thorvg_capi.h#L2350C1-L2350C106).

```c
TVG_API Tvg_Result tvg_animation_get_segment(Tvg_Animation* animation, float* begin, float* end = nullptr); // <-- Use of `nullptr` 
```

### Workaround
For the time being, we have simply removed the default parameter and the library is compiling.

**Comment by tinyjin** [comment]
Thanks for reporting.

We've noticed that the default parameter as a C++ feature harms the C API.
We will soon send a PR for removing this.

**Comment by hermet** [comment]
@andyf-canva Hello, just noticed you have a swift binding(https://github.com/andyf-canva/thorvg-swift). Probably, we can discuss it to upstream in, Please let us know if you'd like to contribute. Thanks.

**Comment by hermet** [comment]
@andyf-canva the fix will be applied in v0.13.3. thanks.

**Comment by andyf-canva** [comment]
@tinyjin @hermet thanks so much for such a quick response to this issue. ❤️ Will pull in the changes when `v0.13.3` is released.

@hermet regarding contributing the swift binding to the upstream repo, we would be delighted to. I do have to double-check our open source contribution policy first, but once I get the clear on that we can move forward. 

What kind of appetite would you have for the code we currently have? I was planning to clean up the Swift API we've written to make it a little bit more general purpose, but keen to hear what you think of it in its current state so we can collaborate on how best to contribute it to the upstream repo. Also, let me know if you'd like to move this discussion to a separate thread.
